src/expr.rs

```diff
@@ -710,7 +710,13 @@ impl Rewrite for ast::Stmt {
                 };
 
                 let shape = shape.sub_width(suffix.len())?;
-                format_expr(ex, ExprType::Statement, context, shape).map(|s| s + suffix)
+                let expr_type =
+                    if stmt_is_expr(&self) && context.snippet(self.span).starts_with("if ") {
```

Comment:
This is a bit hacky - you can inspect the `ast::ExprKind` to see whether the given expression is `if` or not.